pkg/utils/managedresources/managedresources.go

```diff
@@ -186,20 +189,43 @@ func deployManagedResource(ctx context.Context, secret *builder.Secret, managedR
 	return nil
 }
 
-// Delete deletes the managed resource and its secret with the given name in the given namespace.
-func Delete(ctx context.Context, client client.Client, namespace string, name string, secretNameWithPrefix bool) error {
+// Delete deletes the managed resource and its secrets with the given name in the given namespace.
+func Delete(ctx context.Context, c client.Client, namespace string, name string, secretNameWithPrefix bool) error {
 	secretName := SecretName(name, secretNameWithPrefix)
 
-	if err := builder.NewManagedResource(client).
-		WithNamespacedName(namespace, name).
-		Delete(ctx); err != nil {
-		return fmt.Errorf("could not delete managed resource '%s/%s': %w", namespace, name, err)
+	mr := &resourcesv1alpha1.ManagedResource{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: namespace}}
+	mrKey := client.ObjectKeyFromObject(mr)
+	secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: namespace}}
```

Comment:
Can we move this to the `if err != nil && apierrors.IsNotFound(err)` block?

Reply:
I am not sure that we can. In the most recent version of this file the variables are used also outside the scope of the said `if` statement